src/ray/util/event.cc

```diff
@@ -15,6 +15,93 @@
 #include "ray/util/event.h"
 
 namespace ray {
+///
+/// LogEventReporter
+///
+LogEventReporter::LogEventReporter(rpc::Event_SourceType source_type,
+                                   std::string &log_dir, bool force_flush,
+                                   int rotate_max_file_size, int rotate_max_file_num)
+    : log_dir_(log_dir),
+      force_flush_(force_flush),
+      rotate_max_file_size_(rotate_max_file_size),
+      rotate_max_file_num_(rotate_max_file_num) {
+  RAY_CHECK(log_dir_ != "");
+  if (log_dir_.back() != '/') {
+    log_dir_ += '/';
+  }
+  boost::filesystem::create_directories(log_dir_);
```

Comment:
Let's remove boost::filesystem from dependencies and share our log directory here. I don't think there's any situation we'd like to have a separate directory only for events.

Reply:
So your opinion is that we don't need to create the log folder. This operation should be placed outside the framework?
When the log reporter detected this log folder is not created, It means that the log reporter would stop working.

I have a question about which framework would responsible for creating the log folder(Or just the same as the RAY_LOG log_dir)

Reply:
What I am saying is we can just write event logs to the same directory we record log files /tmp/ray/session_*, where RAY_LOG writes log files. I just couldn’t think of any scenario these logs should be recorded in any separate dir from there.

Reply:
OK, I would remove the boost filesystem framework.